clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/LoggingClientBuilder.java

```diff
@@ -74,14 +79,19 @@ private <T> T getProcessor(Class<T> clazz) {
 
   /**
    * Provides a {@link LoggingClientBuilder} with default {@link LogProcessor}s; {@link
-   * ValidationProcessor}, {@link RuntimeInfoProcessor}, and {@link RemoteProcessor}. }
+   * ValidationProcessor}, {@link RuntimeInfoProcessor}, {@link RemoteProcessor}, and when required,
+   * {@link JustificationProcessor}. }
    */
   public LoggingClientBuilder withDefaultProcessors() {
-    return withValidationProcessor()
+    withValidationProcessor()
         .withFilteringProcessor()
         .withRuntimeInfoProcessor()
         .withDefaultLogBackends()
         .withLabelProcessor();
+    if (auditLoggingConfiguration.isJustificationRequired()) {
```

Comment:
Should be the `if` logic in the `withJustificationProcessor` method? Similar to `withDefaultLogBackends`

Reply:
I was hesitating between what you said and what I did here, my reason was that withJustificationProcessor sounds more like it will add justification processor, but I am fine with either.